### PR TITLE
fix(issues) Constrain environment options when a project is forced

### PR DESCRIPTION
**Pull request by markstory**
When a project is forced we should only show the environments for that project as the forced project overrules the project selection.

Fixes ISSUE-335

### CODE REVIEW

**Review by lynnagara — COMMENTED** [review]
It's possible to get into some odd scenarios with this, see https://www.dropbox.com/s/2ergmk24107c2mn/environments.mov?dl=0

Is the current selection supposed to get completely dropped in those cases, and can we prevent some of those intermediate states that get rendered?

**Comment by markstory** [comment]
@lynnagara Looks like you had environments selected that aren't applicable to the issue you navigate to. While the invalid environments should be removed, the valid ones should stay. I'll get that fixed.

File: src/sentry/static/sentry/app/components/organizations/multipleEnvironmentSelector.jsx [hunk 1]
```diff
@@ -79,7 +79,6 @@ class MultipleEnvironmentSelector extends React.PureComponent {
   replaceSelected(newSelection) {
     this.setState({selectedEnvs: new Set(newSelection)});
     this.props.onChange(newSelection);
-    this.props.onUpdate();
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/organizations/multipleEnvironmentSelector.jsx]: This was causing the GlobalSelectionHeader state to be wiped which was causing the flickering.

**Review by lynnagara — APPROVED** [review]
Looks good to me

**Pull request merged**